libs/x-content/src/main/java/org/elasticsearch/xcontent/ParseField.java

```diff
@@ -31,6 +31,8 @@ public class ParseField {
 
     private static final String[] EMPTY = new String[0];
 
+    private final SerializableString serializableName;
+
     private ParseField(
         String name,
```

Comment:
@mosche we could do that, by introducing a `SerializableString#getValue` (or even overriding `toString`, why not?) to return the original value (name) and use that inside ParseField, where needed. Is that what you are meaning?

Reply:
Yes 👍 Or alternatively, given `value` is final / immutable, it could also just be publicly visible.
And it's probably necessary to implement hashCode / equals then.